packages/experiment-core/src/evaluation/evaluation.ts

```diff
@@ -69,28 +69,35 @@ export class EvaluationEngine {
         return undefined;
       }
     }
-    // Outer list logic is "or" (||)
-    for (const conditions of segment.conditions) {
-      let match = true;
-      for (const condition of conditions) {
-        match = this.matchCondition(target, condition);
-        if (!match) {
-          break;
-        }
-      }
-      // On match, bucket the user.
-      if (match) {
-        const variantKey = this.bucket(target, segment);
-        if (variantKey !== undefined) {
-          return flag.variants[variantKey];
-        } else {
-          return undefined;
-        }
+
+    const match = this.evaluateConditions(target, segment.conditions);
+
+    // On match, bucket the user.
+    if (match) {
+      const variantKey = this.bucket(target, segment);
+      if (variantKey !== undefined) {
+        return flag.variants[variantKey];
+      } else {
+        return undefined;
       }
     }
+
     return undefined;
   }
 
+  public evaluateConditions(
+    target: EvaluationTarget,
+    conditions: EvaluationCondition[][],
+  ): boolean {
+    // Outer list logic is "or" (||)
+    return conditions.some((innerConditions) => {
```

Comment:
Avoid language specific syntactic sugar wherever possible. Prefer for loops to make this code easier to maintain across different languages.